File: exemplo.py
#!/usr/bin/env python3
"""Popula o banco com lancamentos de exemplo (opcional).

    python3 exemplo.py            # usa o mes atual
    python3 exemplo.py 2025-08    # usa o mes informado
"""

import sys
from datetime import date

import api
import db

EXEMPLOS = [
    # (pasta, conta, tipo, dia, descricao, valor em reais)
    ("Corretor", "Comissão de venda", 28, "Comissão de venda", 2500.00),
    ("Corretor", "Comissão de venda", 18, "Comissão de venda", 1800.00),
    ("Corretor", "Comissão de locação", 10, "Comissão de locação", 8200.00),
    ("Corretor", "Combustível", 27, "Combustível", 180.00),
    ("Corretor", "Combustível", 12, "Combustível", 1020.00),
    ("Corretor", "Marketing", 25, "Marketing", 650.00),
    ("Corretor", "Alimentação", 22, "Alimentação", 420.00),
    ("Corretor", "Internet/Telefone", 20, "Internet/Telefone", 280.00),
    ("Corretor", "Outros", 15, "Outros", 700.00),
    ("Barbearia", "Serviços (cortes e barba)", 28, "Serviços (cortes e barba)", 320.00),
    ("Barbearia", "Serviços (cortes e barba)", 18, "Serviços (cortes e barba)", 290.00),
    ("Barbearia", "Serviços (cortes e barba)", 8, "Serviços (cortes e barba)", 7140.00),
    ("Barbearia", "Venda de produtos", 14, "Venda de produtos", 1000.00),
    ("Barbearia", "Produtos (pomadas, etc.)", 27, "Produtos", 180.00),
    ("Barbearia", "Produtos (pomadas, etc.)", 5, "Reposicao de estoque", 800.00),
    ("Barbearia", "Aluguel", 25, "Aluguel", 750.00),
    ("Barbearia", "Energia/Água", 22, "Energia/Água", 420.00),
    ("Barbearia", "Internet/Telefone", 20, "Internet/Telefone", 310.00),
    ("Barbearia", "Outros", 16, "Outros", 520.00),
    ("Casa", "Salário", 18, "Salário", 5000.00),
    ("Casa", "Moradia (aluguel/cond.)", 28, "Aluguel", 1200.00),
    ("Casa", "Contas (água, luz, internet)", 27, "Conta de luz", 280.00),
    ("Casa", "Contas (água, luz, internet)", 22, "Internet", 400.00),
    ("Casa", "Alimentação", 25, "Mercado", 520.00),
    ("Casa", "Transporte", 21, "Transporte", 420.00),
    ("Casa", "Outros", 12, "Outros", 600.00),
]


def main():
    mes = sys.argv[1] if len(sys.argv) > 1 else date.today().strftime("%Y-%m")
    db.init()
    with db.cursor() as cur:
        cur.execute("SELECT c.id, c.nome, p.nome AS pasta FROM contas c JOIN pastas p ON p.id = c.pasta_id")
        contas = {(r["pasta"], r["nome"]): r["id"] for r in cur.fetchall()}
    criados = 0
    for pasta, conta, dia, descricao, valor in EXEMPLOS:
        conta_id = contas.get((pasta, conta))
        if not conta_id:
            print(f"  ! conta nao encontrada: {pasta} / {conta}")
            continue
        api.criar_lancamento({
            "conta_id": conta_id, "valor": valor,
            "data": f"{mes}-{dia:02d}", "descricao": descricao,
        })
        criados += 1
    print(f"{criados} lancamentos de exemplo criados em {api.rotulo_mes(mes)}.")


if __name__ == "__main__":
    main()

File: scripts/exemplo.ts
/**
 * Popula o banco com lançamentos de exemplo (opcional).
 *
 *   npm run exemplo            # usa o mês atual
 *   npm run exemplo 2025-08    # usa o mês informado
 */

import { db, prepararBanco } from "../api/_db.js";
import { criarLancamento, hojeISO, rotuloMes, validarMes } from "../api/_servico.js";

const EXEMPLOS: Array<[string, string, number, string, number]> = [
  ["Corretor", "Comissão de venda", 28, "Comissão de venda", 2500],
  ["Corretor", "Comissão de venda", 18, "Comissão de venda", 1800],
  ["Corretor", "Comissão de locação", 10, "Comissão de locação", 8200],
  ["Corretor", "Combustível", 27, "Combustível", 180],
  ["Corretor", "Combustível", 12, "Combustível", 1020],
  ["Corretor", "Marketing", 25, "Marketing", 650],
  ["Corretor", "Alimentação", 22, "Alimentação", 420],
  ["Corretor", "Internet/Telefone", 20, "Internet/Telefone", 280],
  ["Corretor", "Outros", 15, "Outros", 700],
  ["Barbearia", "Serviços (cortes e barba)", 28, "Serviços (cortes e barba)", 320],
  ["Barbearia", "Serviços (cortes e barba)", 18, "Serviços (cortes e barba)", 290],
  ["Barbearia", "Serviços (cortes e barba)", 8, "Serviços (cortes e barba)", 7140],
  ["Barbearia", "Venda de produtos", 14, "Venda de produtos", 1000],
  ["Barbearia", "Produtos (pomadas, etc.)", 27, "Produtos", 180],
  ["Barbearia", "Produtos (pomadas, etc.)", 5, "Reposição de estoque", 800],
  ["Barbearia", "Aluguel", 25, "Aluguel", 750],
  ["Barbearia", "Energia/Água", 22, "Energia/Água", 420],
  ["Barbearia", "Internet/Telefone", 20, "Internet/Telefone", 310],
  ["Barbearia", "Outros", 16, "Outros", 520],
  ["Casa", "Salário", 18, "Salário", 5000],
  ["Casa", "Moradia (aluguel/cond.)", 28, "Aluguel", 1200],
  ["Casa", "Contas (água, luz, internet)", 27, "Conta de luz", 280],
  ["Casa", "Contas (água, luz, internet)", 22, "Internet", 400],
  ["Casa", "Alimentação", 25, "Mercado", 520],
  ["Casa", "Transporte", 21, "Transporte", 420],
  ["Casa", "Outros", 12, "Outros", 600],
];

const mes = validarMes(process.argv[2]) ?? hojeISO().slice(0, 7);
await prepararBanco();

const { rows } = await db().execute(
  "SELECT c.id, c.nome, p.nome AS pasta FROM contas c JOIN pastas p ON p.id = c.pasta_id",
);
const contas = new Map(rows.map((r) => [`${r.pasta}|${r.nome}`, Number(r.id)]));

let criados = 0;
for (const [pasta, conta, dia, descricao, valor] of EXEMPLOS) {
  const contaId = contas.get(`${pasta}|${conta}`);
  if (!contaId) {
    console.log(`  ! conta não encontrada: ${pasta} / ${conta}`);
    continue;
  }
  await criarLancamento({
    conta_id: contaId,
    valor,
    data: `${mes}-${String(dia).padStart(2, "0")}`,
    descricao,
  });
  criados += 1;
}
console.log(`${criados} lançamentos de exemplo criados em ${rotuloMes(mes)}.`);
